syntax = "proto3";

package auth;

// =======================
//  REQUEST MESSAGES
// =======================

message LoginRequest {
  string username = 1;
  string password = 2;
}

message RefreshRequest {
  string refresh_token = 1;
}

message LogoutRequest {
  string refresh_token = 1;
}

// =======================
//  RESPONSE MESSAGES
// =======================

message Token {
  string access_token = 1;
  string token_type = 2;
  string role = 3;
  string refresh_token = 4; // included because your login endpoint uses it
}

message LogoutResponse {
  bool ok = 1;
}

// =======================
//  SERVICE DEFINITION
// =======================

service AuthService {
  rpc Login(LoginRequest) returns (Token);
  rpc Refresh(RefreshRequest) returns (Token);
  rpc Logout(LogoutRequest) returns (LogoutResponse);
}
